src/js_realm.hpp

```diff
@@ -537,6 +541,84 @@ void RealmClass<T>::get_sync_session(ContextType ctx, ObjectType object, ReturnV
     }
 
 }
+
+template<typename T>
+void RealmClass<T>::wait_for_download_completion(ContextType ctx, FunctionType, ObjectType this_object, size_t argc, const ValueType arguments[], ReturnValue &return_value) {
+    validate_argument_count(argc, 2);
+    auto config_object = Value::validated_to_object(ctx, arguments[0]);
+    auto callback_function = Value::validated_to_function(ctx, arguments[1]);
+
+    ValueType sync_config_value = Object::get_property(ctx, config_object, "sync");
+    if (!Value::is_undefined(ctx, sync_config_value))
+    {
+        realm::Realm::Config config;
+        static const String encryption_key_string = "encryptionKey";
+        ValueType encryption_key_value = Object::get_property(ctx, config_object, encryption_key_string);
```

Comment:
no its not. This is always done before the call to parse the config object. This is taken from here: 
https://github.com/realm/realm-js/blob/master/src/js_realm.hpp#L349-L357

I have included it because the end result should be valid realm::Realm::Config config; object the sync properties.